File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int left = 0, curSum = 0, minLen = Integer.MAX_VALUE;

        for(int right = 0; right<nums.length; right++){
            curSum += nums[right];

            while(curSum>=target){
                minLen = Math.min(minLen, right-left+1);
                curSum -= nums[left];
                left++;
            }
        }
        return minLen == Integer.MAX_VALUE ? 0 : minLen;
    }
}

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.swift
class Solution {
    func minSubArrayLen(_ target: Int, _ nums: [Int]) -> Int {
        var left = 0
        var curSum = 0
        var minLen = Int.max

        for right in 0..<nums.count {
            curSum += nums[right]

            while curSum >= target {
                minLen = min(minLen, right-left+1)
                curSum-=nums[left]
                left += 1
            }
        }
        return minLen == Int.max ? 0 : minLen  
    }
}